File: 230A.py
def solve():
    lst = []
    line = input()
    data = line.split()
    s = int(data[0])
    n = int(data[1])
    for i in range(0,n):
        line = input()
        data = line.split()
        tup = (int(data[0]),int(data[1]))
        lst.append(tup)
    lst.sort()
    #print(lst)
    for t in lst:
        x= t[0]
        y= t[1]
        if(x>=s):
            print("NO")
            return 0
        else:
            s += y
    print("YES")
    return 0

solve()

File: 230A.cpp
#include <bits/stdc++.h>
using namespace std;

int solve()
{
  int k, n;
  cin>>k>>n;
  vector<pair<int, int>>vec(n);
  for(int i=0; i<n; i++)
  {
    cin>>vec[i].first>>vec[i].second;
  }
  sort(vec.begin(), vec.end());
  for(int i=0; i<n; i++)
  {
    if(vec[i].first < k)
    {
        k+= vec[i].second;
    }
    else
    {
        cout<<"NO\n";
        return 0;
    }
  }
  cout<<"YES\n";
  return 0;  
}


int main()
{
  int tc = 1;
  //cin>>tc;
  while(tc--)
  {
    solve();
  }

}